lib/nxt_pipeline/pipeline.rb

```diff
@@ -46,10 +46,13 @@ def step(type = nil, **opts, &block)
 
     def execute(arg, &block)
       reset_log
+      before_execute_callback.call(arg) if before_execute_callback.respond_to?(:call)
       configure(&block) if block_given?
-      steps.inject(arg) do |argument, step|
+      result = steps.inject(arg) do |argument, step|
```

Comment:
I think I would simply remember the result in an instance variable and then yield self and args to the callbacks.

Reply:
As discussed I changed it to yield `self` to the callbacks for flexibility.